src/authenticator.ts

```diff
@@ -89,25 +98,30 @@ export default class Authenticator {
       access_token: newAccessToken,
       token_type: "bearer",
       expires_in: TOKEN_EXPIRY,
-      refresh_token: newRefreshToken,
+      refresh_token: newRefreshToken.value,
     });
   }
 
-  private generateAccessToken(options: AuthenticateOptions): string {
+  generateAccessToken(options: AuthenticateOptions): TokenWithExpire {
     let now = Math.floor(Date.now() / 1000);
+    let expire = now + TOKEN_EXPIRY - TOKEN_LEEWAY;
 
     let claims = {
       app: this.appId,
       iss: this.appKeyId,
       iat: now - TOKEN_LEEWAY,
-      exp: now + TOKEN_EXPIRY - TOKEN_LEEWAY,
+      exp: expire,
       sub: options.userId,
+      ...options.serviceClaims,
     };
 
-    return jwt.sign(claims, this.appKeySecret);
+    return {
+      token: jwt.sign(claims, this.appKeySecret),
+      expire: expire,
```

Comment:
I don't really see how it can not be dependant on the clock. The client needs to figure out when the token has expired. How can it do this without checking the current time?